packages/core/src/RenderPipeline/BasicRenderPipeline.ts

```diff
@@ -140,17 +145,25 @@ export class BasicRenderPipeline {
     pass.preRender(camera, this._opaqueQueue, this._alphaTestQueue, this._transparentQueue);
 
     if (pass.enabled) {
-      const rhi = camera.scene.engine._hardwareRenderer;
+      const { engine, scene } = camera;
+      const { background } = scene;
+      const rhi = engine._hardwareRenderer;
       const renderTarget = camera.renderTarget || pass.renderTarget;
       rhi.activeRenderTarget(renderTarget, camera);
       renderTarget?._setRenderTargetFace(cubeFace);
-      rhi.clearRenderTarget(camera.engine, pass.clearMode, pass.clearParam);
+      const clearFlags = pass.clearFlags ?? camera.clearFlags;
+      if (clearFlags !== CameraClearFlags.None) {
+        rhi.clearRenderTarget(camera.engine, clearFlags, background.color);
+      }
 
       if (pass.renderOverride) {
         pass.render(camera, this._opaqueQueue, this._alphaTestQueue, this._transparentQueue);
       } else {
         this._opaqueQueue.render(camera, pass.replaceMaterial, pass.mask);
         this._alphaTestQueue.render(camera, pass.replaceMaterial, pass.mask);
+        if (background.mode === BackgroundMode.Sky && background.sky) {
```

Comment:
`background.sky` is always have, you should check `mesh` and `material` are both OK！
for example add internal fnction to `Sky`:
```typescript
_isVaild():booean
{
  return material&&mesh;
}
```

Reply:
+1